synapse/config/server.py

```diff
@@ -288,6 +289,20 @@ def read_config(self, config, **kwargs):
                 public_baseurl += "/"
         self.public_baseurl = public_baseurl
 
+        # check that public_baseurl is valid
+        try:
+            splits = urllib.parse.urlsplit(self.public_baseurl)
+        except Exception as e:
+            raise ConfigError(f"Unable to parse URL: {e}", ("public_baseurl",))
```

Comment:
I wonder if this has minor potential to cause previously working configs to break (if somehow the generated `public_baseurl` from `server_name` creates an invalid URL). Looks like we already validate that it above though so should be OK.

Reply:
I think it should be ok.

I feel like if `server_name` did generate a bad `pubic_baseurl`, that must be a pretty broken setup anyway.

Reply:
That's fair! 👍